### PR TITLE
front-end and back-end project

### PR DESCRIPTION
**Pull request by J2ZROMERO**
- Apply technical knowledge and skills gained in previous modules in a complex project.
- Understand the pros and cons of different approaches of connecting Ruby on Rails back-end with React front-end.
- Understand principles of Ruby on Rails and React frameworks.
- Apply Ruby best practices and language style guides in code.
- Apply RoR best practices and language style guides in code (e.g. thin controllers).
- Apply JavaScript best practices and language style guides in code.
- Apply React best practices and language style guides in code.
- Learn about and practice giving constructive feedback to teammates.
- Perform a code review for a team member.
- Use the "Review change" feature from GitHub.
- Plan a 2+ week project with no interim Microverse-set milestones and submit it on time.
- Apply knowledge of setting working agreements to set group project teams up for success.
- Independently implement best practices for group agreements to improve teamwork in larger group projects.
- Demonstrate ability to apply best practices of communication for resolving teamwork challenges.
- Understand that respect is the foundation of strong relationship-building with teammates.
- Show up throughout group projects as a reliable and committed team member who communicates and manages your time effectively.
- Recognize the importance of investing in good working relationships with teammates.
- Use an empty Kanban board to manage tasks with a team and own time on the project.

[Live view](https://dev--remarkable-wisp-da7afd.netlify.app/)
[FRONTEND PULL REQUEST](https://github.com/vicmaburrito/online-class-frontend/pull/10)
Integrants: @vicmaburrito @MahdiAghaali @Gleennkar @J2ZROMERO

### CODE REVIEW

File: spec/requests/api/v1/schedules_spec.rb [hunk 1]
```diff
@@ -0,0 +1,138 @@
+require 'swagger_helper'
+
+RSpec.describe 'api/v1/schedules', type: :request do
+  before(:each) do
+    @user = User.create(username: 'juan', email: 'email@hotmail.com', password: '153624')
+    @city = City.create(city_name: 'Brasil')
+    @teacher = Teacher.create(name: 'Carlos', last_name: 'perez', degree: 'mastery')
```
Comment (mcrd25) [inline spec/requests/api/v1/schedules_spec.rb:7]: I get this error: 
image

I get this error whenever the Teacher model is referenced in tests. If you'd need further help with identifying the issue, feel free to reach out to me and may be able to give you 15 minutes to help solve. Can reach out through slack or email. (Details in github profile)

File: spec/models/city_spec.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe City, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (mcrd25) [inline spec/models/city_spec.rb:4]: In general, you are missing unit tests

File: test/controllers/cities_controller_test.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  end
+
+  test 'should get edit' do
+    get cities_edit_url
+    assert_response :success
+  end
+
+  test 'should get create' do
+    get cities_create_url
+    assert_response :success
```
Comment (mcrd25) [inline test/controllers/cities_controller_test.rb:26]: We really prefer RSPEC tests

File: app/controllers/api/v1/schedules_controller.rb [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+      render json: @schedules, status: :created
+    else
+      render json: @schedules.errors, status: :unprocessable_entity
+    end
+  end
+
+  # PATCH/PUT schedules/1 or schedules/1.json
+  def update
+    if @schedules.update(schedule_params)
+      render json: @schedules
```
Comment (mcrd25) [inline app/controllers/api/v1/schedules_controller.rb:30]: [OPTIONAL] consider adding successful update status as well, as seen with create. Makes actions consistent. Try this for all controllers

File: app/controllers/api/v1/schedules_controller.rb [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    else
+      render json: @schedules.errors, status: :unprocessable_entity
+    end
+  end
+
+  # DELETE /schedules/1 or /schedules/1.json
+  def destroy
+    @schedules.destroy
```
Comment (mcrd25) [inline app/controllers/api/v1/schedules_controller.rb:38]: What if it's not successfully destroyed

**Review by mcrd25 — CHANGES_REQUESTED** [review]
Hi  Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- good design ✔️ 
-  tests present ✔️ 
- restful routes implemented correctly ✔️ 
- has Documentation ✔️ 

## Required Changes ♻️
_Check the comments under the review._
- fix rspec error with Teacher model
- there are typos in your README
- Mention Documentation in README
- Don't forget that the application needs to look exactly like design with exception of photos, titles and descriptions
- Is it that I do not needto review Front-End code? I see no reference to it in PR or any relevant PR to review

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
-  [OPTIONAL] Check controllers and see how you can make implementtion of actions more consistent


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Meltrust — APPROVED** [review]
_note: This review outcome applies only to the backend project, as you did the previous reviewer's requirements.  In order to see the frontend project review comment's go to its [PR](https://github.com/vicmaburrito/online-class-frontend/pull/10#pullrequestreview-1345477896)._


#### Hi team,


# **Wow, you did it 🎉**
Brilliant
## Thank you for the changes implemented 💪 🥇 ㊗️ 
## You are about to finish the Microverse program.  You have come a long way!!!




### Good luck in the software industry!!  I'll see you there. ✨
### **Congratulations!!!!!!**  🎉
applause

### To highlight

- Tests have been corrected✔️ 
- Unit tests have been added✔️ 
- Controllers have been fixed✔️ 



Cheers and Happy coding!👏👏👏


------

**Pull request merged**